bit_manipulation/same_number_of_bits.cpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file
+ *
+ * @brief Implementation to [Next higher number with same number of set
+ * bits](https://www.geeksforgeeks.org/next-higher-number-with-same-number-of-set-bits/)
+ * @details
+ * We have given a number let's say n, find next number with the same number of
+ * 1 bit in its binary representation.
+ *
+ * @author [Kunal Kumar](https://github.com/kunalkumar37)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace bit_manipulation
+ * @brief Bit manipulation algorithms
+ */
+namespace bit_manipulation {
+/**
+ * @namespace same_number_of_bits
+ * @brief Functions for the [Next higher number with the same number of set
+ * bits](https://www.geeksforgeeks.org/next-higher-number-with-same-number-of-set-bits/)
+ * implementation implementation
+ */
+namespace same_number_of_bits {
+/**
+ * @brief The main function implements same number of bits
+ * @returns this function returns next higher with same number of set bits as n
```

Comment:
Missing `n` parameter documentation. You must document what is the `n` variable for.